### PR TITLE
Profile view controller

### PR DESCRIPTION
**Pull request by ishamoel**

### CODE REVIEW

File: net/NSString+stringIsEmpry.m [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+//  Created by user on 2/19/16.
+//  Copyright © 2016 Admin. All rights reserved.
+//
+
+#import "NSString+stringIsEmpry.h"
+
+@implementation NSString (stringIsEmpty)
+
++ (BOOL) stringIsEmpty:(NSString *)string {
+    if ((NSNull *) string == [NSNull null])
```
Comment (simple-user) [inline net/NSString+stringIsEmpry.m]: How does it work? It's just comparing of two pointers. It's not  a calling overloaded function as in Cpp 
(like bool operator==(SomeType &))

File: net/ProfileViewController.m [hunk 2]
```diff
@@ -54,58 +59,44 @@ - (void)viewDidAppear:(BOOL)animated {
...
-            case MANAGER:
-                [self.systemRole setText:@"MANAGER"];
-                break;
-            case USER:
-                [self.systemRole setText:@"USER"];
-                break;
-            case SUBSCRIBER:
-                [self.systemRole setText:@"SUBSCRIBER"];
-                break;
+    if ([[self backViewController] isKindOfClass:[IssueHistoryViewController class]]) {
```
Comment (dscherb) [inline net/ProfileViewController.m:70]: I believe  you forgot "!" before in condition

File: net/ProfileViewController.m [hunk 3]
```diff
@@ -54,58 +59,44 @@ - (void)viewDidAppear:(BOOL)animated {
...
+    
+        if ([CurrentItems sharedItems].user && (self.userID == [CurrentItems sharedItems].user.userId)) {
+            [self showUserProfile];
+        }
+        else {
+            [self requestUserDetailsByID:self.userID updateScreenWithHandler:^(User *user){
+                
+                dispatch_async(dispatch_get_main_queue(), ^{
+                    
+                    [self.userLogin setText:[NSString stringWithFormat:@"@%@", user.login]];
```
Comment (dscherb) [inline net/ProfileViewController.m]: Why don't you use showUserProfile method here?

File: net/ProfileViewController.m [hunk 4]
```diff
@@ -145,7 +160,13 @@ - (void) revealAllViews {
         [self.labelSystemRole setHidden:NO];
         [self.systemRole setHidden:NO];
         
-        if (self.userID == [CurrentItems sharedItems].user.userId) {
+        if ([[self backViewController] isKindOfClass:[IssueHistoryViewController class]]) {
```
Comment (dscherb) [inline net/ProfileViewController.m:155]: Some sing wrong with class names. You check if backViewController is a IssueHistoryViewController, and show current user info. It should be something like CurrentUserViewController. Isn't it?

File: net/ProfileViewController.m [hunk 5]
```diff
@@ -155,7 +176,7 @@ - (void) revealAllViews {
     });
 }
 
-- (void) requestAvatarWithName: (NSString *) avatarName {
+- (void) requestAvatarWithName: (NSString const * const) avatarName {
```
Comment (dscherb) [inline net/ProfileViewController.m:171]: NSString is not mutable by his nature. Why you need const here?

File: net/ProfileViewController.m [hunk 6]
```diff
@@ -189,7 +210,9 @@ - (void) requestUserDetailsByID: (NSUInteger) ID updateScreenWithHandler:(void(^
                                             NSDictionary *userData = [NSJSONSerialization JSONObjectWithData:data
                                                                                                      options:0
                                                                                                        error:NULL];
-                                            self.avatarImageURL = [NSString stringWithFormat:@"%@", userData[@"AVATAR"]];
+                                            if (userData[@"AVATAR"] != [NSNull null])
+                                                self.avatarImageURL = [[NSString alloc] initWithString:userData[@"AVATAR"]];
```
Comment (dscherb) [inline net/ProfileViewController.m]: You need to use weak pointer on to self inside block

**Pull request closed without merging**